src/v/kafka/client/consumer.h

```diff
@@ -115,18 +116,23 @@ class consumer final : public ss::enable_lw_shared_from_this<consumer> {
               r,
               _coordinator->id());
             return _coordinator->dispatch(std::move(r))
-              .then([this](response_t res) {
+              .then([this, req{std::move(req)}](response_t res) {
                   kclog.debug(
                     "Consumer: {}: {} res: {}, coordinator {}",
                     *this,
                     api_t::name,
                     res,
                     _coordinator->id());
-                  return res;
+                  return maybe_process_response_errors(
+                    std::move(req), std::move(res));
```

Comment:
Compile error in `consumer::sync` as a result of this change but I do not know why ...
```bash
ninja: Entering directory `/home/nyalia/redpanda/vbuild/debug/clang'
[23/90] Building CXX object src/v/kafka/client/CMakeFiles/v_kafka_client.dir/consumer.cc.o
FAILED: src/v/kafka/client/CMakeFiles/v_kafka_client.dir/consumer.cc.o 
/usr/bin/ccache /home/nyalia/redpanda/vbuild/llvm/install/bin/clang++ -DBASE64_STATIC_DEFINE -DBOOST_ASIO_HAS_STD_INVOKE_RESULT -DFMT_LOCALE -DSEASTAR_API_LEVEL=6 -DSEASTAR_BROKEN_SOURCE_LOCATION -DSEASTAR_DEBUG -DSEASTAR_DEBUG_SHARED_PTR -DSEASTAR_DEFAULT_ALLOCATOR -DSEASTAR_SCHEDULING_GROUPS_COUNT=16 -DSEASTAR_SHUFFLE_TASK_QUEUE -DSEASTAR_TYPE_ERASE_MORE -DXXH_PRIVATE_API -DZSTD_STATIC_LINKING_ONLY -I/home/nyalia/redpanda/src/v -I/home/nyalia/redpanda/vbuild/debug/clang/src/v -isystem /home/nyalia/redpanda/vbuild/debug/clang/rp_deps_install/include -isystem /home/nyalia/redpanda/vbuild/debug/clang/rp_deps_install/include/hdr -isystem /home/nyalia/redpanda/vbuild/debug/clang/rp_deps_install/include/libxml2 -fPIC -march=westmere -fuse-ld=lld -mllvm -inline-threshold=2500 -Wno-unused-lambda-capture -Wno-unused-command-line-argument -stdlib=libc++ -fsanitize=undefined -fsanitize=address  -O0 -g -gdwarf-4 -gsplit-dwarf -fPIC -Wall -Wextra -Werror -Wno-missing-field-initializers -U_FORTIFY_SOURCE -DSEASTAR_SSTRING -Werror=unused-result "-Wno-error=#warnings" -fstack-clash-protection -fsanitize=address -fsanitize=undefined -fno-sanitize=vptr -std=c++20 -MD -MT src/v/kafka/client/CMakeFiles/v_kafka_client.dir/consumer.cc.o -MF src/v/kafka/client/CMakeFiles/v_kafka_client.dir/consumer.cc.o.d -o src/v/kafka/client/CMakeFiles/v_kafka_client.dir/consumer.cc.o -c /home/nyalia/redpanda/src/v/kafka/client/consumer.cc
In file included from /home/nyalia/redpanda/src/v/kafka/client/consumer.cc:10:
In file included from /home/nyalia/redpanda/src/v/kafka/client/consumer.h:14:
In file included from /home/nyalia/redpanda/src/v/kafka/client/assignment_plans.h:14:
In file included from /home/nyalia/redpanda/src/v/kafka/protocol/join_group.h:13:
In file included from /home/nyalia/redpanda/src/v/kafka/protocol/errors.h:15:
In file included from /home/nyalia/redpanda/vbuild/llvm/install/bin/../include/c++/v1/string_view:206:
In file included from /home/nyalia/redpanda/vbuild/llvm/install/bin/../include/c++/v1/algorithm:667:
In file included from /home/nyalia/redpanda/vbuild/llvm/install/bin/../include/c++/v1/functional:506:
In file included from /home/nyalia/redpanda/vbuild/llvm/install/bin/../include/c++/v1/__functional/function.h:20:
/home/nyalia/redpanda/vbuild/llvm/install/bin/../include/c++/v1/__memory/allocator_traits.h:298:9: error: no matching function for call to 'construct_at'
        _VSTD::construct_at(__p, _VSTD::forward<_Args>(__args)...);
        ^~~~~~~~~~~~~~~~~~~
/home/nyalia/redpanda/vbuild/llvm/install/bin/../include/c++/v1/__config:826:15: note: expanded from macro '_VSTD'
#define _VSTD std
              ^
/home/nyalia/redpanda/vbuild/llvm/install/bin/../include/c++/v1/memory:885:18: note: in instantiation of function template specialization 'std::allocator_traits<std::allocator<kafka::metadata_response_topic>>::construct<kafka::metadata_response_topic, kafka::metadata_response_topic &, void, void>' requested here
        _Traits::construct(__a, _VSTD::__to_address(__begin2), *__begin1);
                 ^
/home/nyalia/redpanda/vbuild/llvm/install/bin/../include/c++/v1/vector:1006:12: note: in instantiation of function template specialization 'std::__construct_range_forward<std::allocator<kafka::metadata_response_topic>, kafka::metadata_response_topic *, kafka::metadata_response_topic *>' requested here
    _VSTD::__construct_range_forward(this->__alloc(), __first, __last, __tx.__pos_);
           ^
/home/nyalia/redpanda/vbuild/llvm/install/bin/../include/c++/v1/vector:1158:9: note: in instantiation of function template specialization 'std::vector<kafka::metadata_response_topic>::__construct_at_end<kafka::metadata_response_topic *>' requested here
        __construct_at_end(__x.__begin_, __x.__end_, __n);
        ^
/home/nyalia/redpanda/src/v/kafka/client/consumer.cc:292:30: note: in instantiation of member function 'std::vector<kafka::metadata_response_topic>::vector' requested here
          auto req_builder = [me{shared_from_this()},
                             ^
/home/nyalia/redpanda/src/v/kafka/client/consumer.cc:306:18: note: in instantiation of function template specialization 'kafka::client::consumer::req_res<(lambda at /home/nyalia/redpanda/src/v/kafka/client/consumer.cc:292:30)>' requested here
          return req_res(std::move(req_builder))
                 ^
/home/nyalia/redpanda/vbuild/llvm/install/bin/../include/c++/v1/__memory/construct_at.h:36:16: note: candidate template ignored: substitution failure [with _Tp = kafka::metadata_response_topic, _Args = <kafka::metadata_response_topic &>]: call to implicitly-deleted copy constructor of 'kafka::metadata_response_topic'
constexpr _Tp* construct_at(_Tp* __location, _Args&& ...__args) {
               ^
1 error generated.
```

Reply:
Make the lambda mutable:  `.then([this, req{std::move(req)}](response_t res) mutable {`

Reply:
Thanks! I forgot about this error